federatedscope/core/sampler.py

```diff
@@ -23,14 +23,19 @@ def change_state(self, indices, state):
         """
         To modify the state of clients (idle or working)
         """
-        if state == 'idle':
-            self.client_state[indices] = 1
-        elif state == 'working':
-            self.client_state[indices] = 0
+        if not isinstance(indices, list):
```

Comment:
Shall we cover np.array here?

Reply:
modified accordingly, besides, we used another state named "unseen"